doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -6,117 +6,397 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-profiles/
+profile update request and endpoint membership in the endpoint groups is re-evaluated and updated to match the new endpoint profile.
+The client-side endpoint profile is unidirectionally synchronized, and thus should not be considered as a means to temporarily store endpoint data in the Kaa cluster. 
+There is no way for the endpoint to retrieve the profile information back from the Kaa cluster. At its start, the application must fill in the current endpoint profile 
+with the up-to-date data and execute corresponding SDK API calls. The endpoint SDK does not persist the profile information over the endpoint reboots. However, 
+it detects profile data changes and submits the new data to the Kaa cluster as a profile update.
 
+<img src="endpoint-profile-generation/ClientSideEndpointProfileGeneration_0_8_0.png">
 
-## Client-side endpoint profile update ##
-When the endpoint profile information changes as the result of the client operation or user's actions, it is the client implementation responsibility to update the profile via the endpoint SDK API calls. The endpoint SDK detects profile changes by comparing the new profile hash against the previously persisted one. Should there be a change, the endpoint profile management module passes it to the Operations server, which in turn updates the endpoint profile information in the database and revises the endpoint groups membership.
-Client-side endpoint profile example
-The following code block provides a simple client-side endpoint profile schema example.
+### Client-side endpoint profile example ###
 
+For the purpose of this guide we will use a fairly abstract client-side profile schema shown below.
 
 ```json
 {
-    "namespace": "org.myproject",
-    "type": "record",
-    "name": "MyClientSideProfile",
-    "fields": [
-        { "name": "country", "type": "string" },
-        { "name": "city", "type": "string" },
-        { "name": "age", "type": "int" },
+    "type":"record",
+    "name":"Profile",
+    "namespace":"org.kaaproject.kaa.schema.sample.profile",
+    "fields":[
+        {
+            "name":"id",
+            "type":"string"
+        },
         {
-            "name": "hobbies",
-            "type": { "type": "array", "items": "string" }
+            "name":"os",
+            "type":{
+                "type":"enum",
+                "name":"OS",
+                "symbols":[
+                    "Android",
+                    "iOS",
+                    "Linux"
+                ]
+            }
+        },
+        {
+            "name":"os_version",
+            "type":"string"
+        },
+        {
+            "name":"build",
+            "type":"string"
         }
     ]
 }
 ```
 
-## The following client-side profile would be compatible with our schema example. ##
+The following client-side profile would be compatible with our schema example.
```

Comment:
I believe, the client-side profile **must** be compatible with the schema. Maybe, it would be better to say that the profile is based on the schema, thus, it is defined by the schema and must follow its 'rules'.